lib/dradis/plugins/mediawiki/filters.rb

```diff
@@ -7,17 +7,19 @@ def query(params={})
       path   = Dradis::Plugins::Mediawiki::Engine.settings.path
       port   = Dradis::Plugins::Mediawiki::Engine.settings.port
       scheme = Dradis::Plugins::Mediawiki::Engine.settings.scheme
+      limit = Dradis::Plugins::Mediawiki::Engine.settings.limit
 
       port   = (scheme == 'https' ? 443 : 80) if port.blank?
 
       begin
         # Parameters required by MediaWiki API
-        # http://localhost/mediawiki-1.21.1/api.php?action=query&prop=revisions&generator=search&gsrwhat=text&gsrsearch=Directory&rvprop=content&format=xml
+        # http://localhost/mediawiki-1.21.1/api.php?action=query&prop=revisions&generator=search&gsrwhat=text&gsrlimit=max&gsrsearch=Directory&rvprop=content&format=xml
         filter_params = {
              action: 'query',
                prop: 'revisions',
           generator: 'search',
             gsrwhat: 'text',
+           gsrlimit: CGI::escape(limit),
```

Comment:
we are not escaping the rest of settings

Reply:
We are escaping the only other user supplied value in this list, `gsrsearch`. That's where I took my queues from, I can't speak to whether it provides any benefit.

Reply:
Had a look at [Rails' `to_query` method](https://github.com/rails/rails/blob/v6.0.2.1/activesupport/lib/active_support/core_ext/object/to_query.rb), and it calls `CGI::escape` on both keys and values. So yes, it seems like escaping `gsrlimit` and `gsrsearch` is superfluous.

For valid limit values (positive integers and `"max"`) it shouldn't make any difference, but I could imagine some search terms currently get double-escaped.